### ISSUE TITLE
Autocomplete showing more specific concepts before more generic

### ISSUE THREAD

**Issue by CaseyTa** [NCATSTranslator/Feedback#448]
Testing on https://ui.ci.transltr.io/

* Typed in "type 2 diabetes", but "type 2 diabetes mellitus" only shows up on the 3rd page of autocomplete results (click "Show More" twice)
* Typed in "heart disorde", but "heart disorder" shows up on the 2nd page of results
* Typed in "alzheime", but "alzheimer disease" shows up on the 2nd page of results. Typing in "alzheimer" (adding in the last 'r') does bring up "alzheimer disease" to the top result
* Typed in "muscular dystroph", but "muscular dystrophy" shows up below 2 other more specific forms, including an "Muscular Dystrohpy, Non-human Animal" (but it is on the 1st page of results at least)
* Typed in "multiple sclerosi", but "multiple sclerosis" shows up 4th, below some unrelated conditions

**Comment by khanspers** [comment]
Same for Charcot-Marie-Tooth Disease:

- Typing in "Charc" and autocompete ONLY lists specific sub-types of this disease (several autocomplete pages), not the parent term (MONDO:0015626). Typing in "Charcot-Marie" and the expected generic term shows up at the top.

**Comment by sstemann** [comment]
seems related to #447

**Comment by dnsmith124** [comment]
@gaurav @Genomewide could you two take a look into these cases? Since the UI is just displaying what the Name Resolver sends to us there's not really anything I can do to adjust these results aside from changing the curie prefix or biolink type we filter by, and I'd want @Genomewide's approval before doing so.

**Comment by gaurav** [comment]
I've moved the first two cases into separate issues, since the place to start with those issues are relatively straightforward. The third one is now fixed (yay!), but the fix for the last two aren't clear to me. We could implement sorting by the shortest preferred name (https://github.com/TranslatorSRI/NameResolution/issues/72) and see if that helps with these issues, but I feel like there's a smarter way to proceed there.
* "type 2 diabetes" still fails badly, although "diabetes type 2" works fine. I'm tracking that at https://github.com/TranslatorSRI/NameResolution/issues/96
* "heart disorde" still does pretty badly. I'm tracking that at https://github.com/TranslatorSRI/NameResolution/issues/97
* "alzheime" now returns [MONDO:0004975 "Alzheimer disease"](http://purl.obolibrary.org/obo/MONDO_0004975) as the best result, probably because of CURIE suffix sorting.
* Typed in "muscular dystroph", but "muscular dystrophy" shows up below 2 other more specific forms, including an "Muscular Dystrophy, Non-human Animal" (but it is on the 1st page of results at least) -- unchanged
* Typed in "multiple sclerosi", but "multiple sclerosis" shows up 4th, below some unrelated conditions -- unchanged
